src/app/clusters/bindings/bindings.cpp

```diff
@@ -20,155 +20,265 @@
 {
-    NodeId nodeId             = commandData.nodeId;
-    GroupId groupId           = commandData.groupId;
-    ClusterId clusterId       = commandData.clusterId;
-    EndpointId remoteEndpoint = commandData.endpointId;
-    EndpointId localEndpoint  = commandPath.mEndpointId;
-    FabricIndex fabricIndex   = commandObj->GetAccessingFabricIndex();
-    EmberBindingTableEntry bindingEntry;
-
-    ChipLogDetail(Zcl, "RX: UnbindCallback");
-
-    if ((groupId != 0 && nodeId != 0) || (groupId == 0 && nodeId == 0))
-    {
-        ChipLogError(Zcl, "Binding: Invalid request");
-        emberAfSendImmediateDefaultResponse(EMBER_ZCL_STATUS_MALFORMED_COMMAND);
-        return true;
-    }
-    if (groupId)
-    {
-        bindingEntry = EmberBindingTableEntry::ForGroup(fabricIndex, groupId, localEndpoint, clusterId);
-    }
-    else
+    switch (path.mAttributeId)
     {
-        bindingEntry = EmberBindingTableEntry::ForNode(fabricIndex, nodeId, localEndpoint, remoteEndpoint, clusterId);
+    case Binding::Attributes::BindingList::Id:
+        return WriteBindingTable(path, decoder);
+    default:
+        break;
     }
+    return CHIP_NO_ERROR;
+}
 
-    uint8_t bindingIndex;
-    if (getBindingIndex(bindingEntry, &bindingIndex) != EMBER_SUCCESS)
+CHIP_ERROR BindingTableAccess::WriteBindingTable(const ConcreteDataAttributePath & path, AttributeValueDecoder & decoder)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    if (!path.IsListOperation() || path.mListOp == ConcreteDataAttributePath::ListOperation::ReplaceAll)
     {
-        emberAfSendImmediateDefaultResponse(EMBER_ZCL_STATUS_NOT_FOUND);
-        return true;
-    }
+        printf("replace all\n");
+        DecodableBindingListType newBindingList;
+
+        ReturnErrorOnFailure(decoder.Decode(newBindingList));
+        ReturnErrorOnFailure(CheckValidBindingList(newBindingList));
+
+        // Remove entries not in the new binding list
+        for (uint8_t i = 0; i < EMBER_BINDING_TABLE_SIZE; i++)
+        {
+            EmberBindingTableEntry entry;
+            emberGetBinding(i, &entry);
+            if (entry.type != EMBER_UNUSED_BINDING && !IsInBindingList(path.mEndpointId, entry, newBindingList))
+            {
 
-    CHIP_ERROR err = BindingManager::GetInstance().UnicastBindingRemoved(bindingIndex);
-    if (err != CHIP_NO_ERROR)
+                if (entry.type == EMBER_UNICAST_BINDING)
+                {
+                    err = BindingManager::GetInstance().UnicastBindingRemoved(i);
+                    if (err != CHIP_NO_ERROR)
+                    {
+                        ChipLogError(Zcl,
+                                     "Binding: Failed to remove pending notification for unicast binding " ChipLogFormatX64
+                                     ": %" CHIP_ERROR_FORMAT,
+                                     ChipLogValueX64(entry.nodeId), err.Format());
+                    }
+                }
+            }
+        }
+        emberClearBinding();
```

Comment:
So this clears out the binding table... but then the code below checks whether things are in the binding table.  That is not right.  Was this tested?